### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by kazumaki**
Create the logic modules for the calculator

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const returnData = data;
+  if (button === '+/-') {
+    returnData.total *= -1;
+    returnData.next *= -1;
+  } else {
+    returnData.total = operate(data.total, data.next, data.operation);
+  }
+
+  return returnData;
+};
```
Comment (bolah2009) [inline src/logic/calculate.js]: This logic is not comprehensive enough.

Think of cases you need to handle, please remember that `calculate` is the only place where the app logic would be. For each click, `calculate` function will be called.

Think of this and make changes accordingly.

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Hi @kazumaki 

Good job with the project.

Can you please merge #1, #2 and #3, then make sure there are no conflicts before continuing with this milestone?
Also, I noticed you have open another PR #5 for Milestone 5. Please, don't do this as the project is in milestones for a reason.

### Please see the comment below for the required changes.

#### :recycle:  Make these changes and submit another code review request. 👍🏾

#### Feel free to reach me on slack if you having issues implementing these features. :relaxed: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Review by fatymahmed — CHANGES_REQUESTED** [review]
Hi @kazumaki,

As mentioned in the previous review, `calculate` will be called for each button click, that means you should cover all cases.

See my comment below
### Please make the required change then request for another review ♻️
_Feel free to reach out to me on slack (Fatima) if you have any questions_

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    returnData.total = operate(data.total || 0, data.next || 0, data.operation);
+    returnData.next = 0;
+    returnData.display = returnData.total;
+  } else if (button === '+/-') {
+    returnData.total *= -1;
+    returnData.next *= -1;
+    returnData.display = returnData.total;
+  } else {
+    returnData.next = data.next ? data.next + button : button;
+    returnData.display = returnData.next;
```
Comment (fatymahmed) [inline src/logic/calculate.js]: - You are only covering cases for `+` `-` `x`  `%` `/` `=` `+/-`, what about when the user clicks on `AC` or `.` or when an operation is done with `0` (like division or multiplication)?

**Review by Oghenebrume50 — APPROVED** [review]
### STATUS: CHANGES REQUIRED

Hi @kazumaki good job working on this,  you have met all the corrections as outlined by the previous reviewers, please merge and move on to the next milestone :+1: 

**RAPHAEL NORIODE** :mask:

**Pull request merged**